backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java

```diff
@@ -436,15 +444,28 @@ protected Map<String, Pair<String, String>> getSharedLocks() {
             // StartVmBackup should handle locks
             return locks;
         }
-        if (!Guid.isNullOrEmpty(getParameters().getImageId()) && getDiskImage() != null) {
+
+        if (!Guid.isNullOrEmpty(getParameters().getImageId()) ||
+                getDiskImage() != null &&
+                        !isThinTemplate(getDiskImage())) {
```

Comment:
so now if we download a disk of a vm that is set with thin-provisioning, we won't lock the vm? something is suspicious here

Reply:
yeah... it's probably not needed